itests/src/test/java/org/apache/unomi/itests/IncrementInterestsIT.java

```diff
@@ -69,168 +68,131 @@
-        incrementAction.setParameter( "eventInterestProperty", "eventProperty::target.properties.interests" );
+        Event event = createEvent(profile, interestsAsMap);
 
-        final Condition condition = new Condition( definitionsService.getConditionType( "eventTypeCondition" ) );
-        condition.setParameter( "eventTypeId", "view" );
+        int eventCode = eventService.send(event);
 
-        final String itemId = UUID.randomUUID().toString();
+        if (eventCode == EventService.PROFILE_UPDATED) {
+            Profile updatedProfile = profileService.save(event.getProfile());
 
-        final Metadata metadata = new Metadata();
-        metadata.setId( itemId );
-        metadata.setName( itemId );
-        metadata.setDescription( itemId );
-        metadata.setEnabled( true );
-        metadata.setScope( "systemscope" );
+            Map<String, Double> interests = (Map<String, Double>) updatedProfile.getProperty("interests");
 
-        final Rule rule = new Rule();
+            Assert.assertEquals(0.5, interests.get(topic.getTopicId()), 0.0);
+            Assert.assertFalse(interests.containsKey("unknown"));
+        } else {
+            Assert.fail("Profile was not updated");
+        }
+    }
 
-        rule.setCondition( condition );
-        List<Action> actions = new ArrayList<>();
-        actions.add(incrementAction);
-        rule.setActions( actions );
-        rule.setMetadata( metadata );
+    @Test
+    public void testAction() throws InterruptedException {
+        Action incrementAction = new Action(definitionsService.getActionType("incrementInterestAction"));
+        incrementAction.setParameter("eventInterestProperty", "eventProperty::target.properties.interests");
 
-        rulesService.setRule( rule );
+        Condition condition = new Condition(definitionsService.getConditionType("eventTypeCondition"));
+        condition.setParameter("eventTypeId", "view");
 
-        keepTrying( "Failed waiting for the creation of the rule for the IncrementInterestsIT test",
-                    () -> rulesService.getRule( rule.getItemId() ), Objects::nonNull, 1000, 100 );
+        String itemId = UUID.randomUUID().toString();
 
-        final Map<String, Double> interestsAsMap = new HashMap<>();
-        interestsAsMap.put( topic.getTopicId(), 50.0 );
-        interestsAsMap.put( "unknown", 10.0 );
+        Metadata metadata = new Metadata();
+        metadata.setId(itemId);
+        metadata.setName(itemId);
+        metadata.setDescription(itemId);
+        metadata.setEnabled(true);
+        metadata.setScope("systemscope");
 
-        final Map<String, Object> properties = new HashMap<>();
+        rule.setCondition(condition);
+        List<Action> actions = new ArrayList<>();
+        actions.add(incrementAction);
+        rule.setActions(actions);
+        rule.setMetadata(metadata);
 
-        properties.put( "interests", interestsAsMap );
+        rulesService.setRule(rule);
+        refreshPersistence();
```

Comment:
The rule is not remove at the end of the tests, should be.

Reply:
it's removed here: https://github.com/apache/unomi/pull/277/files#diff-f06e946def04bc51838d479376e9f0b4f2b074c4a5e01093ec712e327762f90dR84